### PR TITLE
Prepare to release 3.7.4

### PR DESCRIPTION
**Pull request by xiang17**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,96 @@
...
+* Suppress warnings for some safe operations
+  ([#4371](https://github.com/microsoft/ApplicationInsights-Java/pull/4371))
+* Update Gradle to latest version 8.14.3
+  ([#4345](https://github.com/microsoft/ApplicationInsights-Java/pull/4345))
+* Update to OpenTelemetry Java Instrumentation 2.18.1
+  ([#4378](https://github.com/microsoft/ApplicationInsights-Java/pull/4378))
+* Fix a couple more gradle deprecations
+  ([#4367](https://github.com/microsoft/ApplicationInsights-Java/pull/4367))
+* Fix deprecated EnvironmentResourceProvider reference
+  ([#4377](https://github.com/microsoft/ApplicationInsights-Java/pull/4377))
```
Comment (trask) [inline CHANGELOG.md]: I think this is the only user facing PR here (it solves a deprecation warning that was being printed that users saw)

The other one that we are getting fixed in azure-monitor-opentelemetry-autoconfigure that's worth mentioning since reported by user is: https://github.com/microsoft/ApplicationInsights-Java/pull/4314

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
 
+## Version 3.7.4 GA (08/05/2025)
+
+### Bug fixes
+
+* Fix deprecated EnvironmentResourceProvider reference
+  ([#4377](https://github.com/microsoft/ApplicationInsights-Java/pull/4377))
+* Add smoke tests for null messages. This was initiated here but fixed in
+  the other repository's azure-monitor-opentelemetry-autoconfigure package.
+  ([#4314](https://github.com/microsoft/ApplicationInsights-Java/pull/4314))
```
Comment (trask) [inline CHANGELOG.md]:
```suggestion
* Fix null exception message error
  ([https://github.com/Azure/azure-sdk-for-java/pull/46001](https://github.com/Azure/azure-sdk-for-java/pull/46001)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+
```
Comment (trask) [inline CHANGELOG.md:13]:
```suggestion
* Update to OpenTelemetry Java Instrumentation 2.18.1
  ([#4378](https://github.com/microsoft/ApplicationInsights-Java/pull/4378))

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
 # CHANGELOG
 
+## Version 3.7.4 GA (08/05/2025)
+
+### Bug fixes
+
+* Fix deprecated EnvironmentResourceProvider reference
+  ([#4377](https://github.com/microsoft/ApplicationInsights-Java/pull/4377))
+* Fix null exception message error
+  ([https://github.com/Azure/azure-sdk-for-java/pull/46001](https://github.com/Azure/azure-sdk-for-java/pull/46001)
```
Comment (trask) [inline CHANGELOG.md]:
```suggestion
  ([https://github.com/Azure/azure-sdk-for-java/pull/46001](https://github.com/Azure/azure-sdk-for-java/pull/46001))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,16 @@
 # CHANGELOG
 
+## Version 3.7.4 GA (08/05/2025)
+
+### Bug fixes
+
+* Fix deprecated EnvironmentResourceProvider reference
```
Comment (trask) [inline CHANGELOG.md]:
```suggestion
* Update deprecated EnvironmentResourceProvider reference
```
